packages/tools/devtools/devtools-core/src/data-visualization/DataVisualization.ts

```diff
@@ -224,15 +246,18 @@ export class DataVisualizerGraph
 	 * Adds a visualizer node to the collection for the specified
 	 * {@link @fluidframework/shared-object-base#ISharedObject} if one does not already exist.
 	 */
-	private registerVisualizerForSharedObject(sharedObject: ISharedObject): FluidObjectId {
+	private registerVisualizerForSharedObject(
+		sharedObject: ISharedObject | DataObject,
+	): FluidObjectId {
 		if (!this.visualizerNodes.has(sharedObject.id)) {
 			// Create visualizer node for the shared object
-			const visualizationFunction =
-				this.visualizers[sharedObject.attributes.type] ?? visualizeUnknownSharedObject;
+			const visualizationFunction = isDataObject(sharedObject)
+				? visualizeDataObject
+				: (this.visualizers[sharedObject.attributes.type] ?? visualizeUnknownSharedObject);
 
 			const visualizerNode = new VisualizerNode(
-				sharedObject,
-				visualizationFunction,
+				isDataObject(sharedObject) ? (sharedObject as VisualDataObject).root : sharedObject,
```

Comment:
For what it's worth, I don't think introducing the fake `VisualDataObject` type and casting to it here is really any better than casting to `any`. Either way, we're hacking around the fact that `root` is inaccessible (private).

My recommendation: replace the cast to `VisualDataObject` with a cast to `any` (with the appropriate linter disable and comment explaining what we're doing), and add a check to verify that `root` is a `SharedDirectory` as we expect.